code/modules/shuttle/computer.dm

```diff
@@ -217,29 +303,16 @@
 			SSshuttle.moveShuttle(ert.id, dock.id, TRUE)
 			to_chat(usr, SPAN_NOTICE("You begin the launch sequence to [dock]."))
 			return TRUE
-		if("button-push")
-			playsound(loc, get_sfx("terminal_button"), KEYBOARD_SOUND_VOLUME, 1)
-			return FALSE
-		if("open")
-			if(ert.mode == SHUTTLE_CALL || ert.mode == SHUTTLE_RECALL)
-				return TRUE
-			ert.control_doors("open", external_only = TRUE)
-		if("close")
-			if(ert.mode == SHUTTLE_CALL || ert.mode == SHUTTLE_RECALL)
-				return TRUE
-			ert.control_doors("close", external_only = TRUE)
-		if("lockdown")
-			if(ert.mode == SHUTTLE_CALL || ert.mode == SHUTTLE_RECALL)
-				return TRUE
-			ert.control_doors("force-lock", external_only = TRUE)
-		if("lock")
-			if(ert.mode == SHUTTLE_CALL || ert.mode == SHUTTLE_RECALL)
-				return TRUE
-			ert.control_doors("lock", external_only = TRUE)
-		if("unlock")
-			if(ert.mode == SHUTTLE_CALL || ert.mode == SHUTTLE_RECALL)
-				return TRUE
-			ert.control_doors("unlock", external_only = TRUE)
+		if("launch_home")
+			if(!must_launch_home)
+				return
+
+			if(ert.mode != SHUTTLE_IDLE)
+				to_chat(ui.user, SPAN_WARNING("Unable to return home."))
+				balloon_alert(ui.user, "can't go home!")
+				return
+
+			launch_home()
```

Comment:
```suggestion
			launch_home()
			return TRUE
```
This should still return TRUE so ideally the UI updates immediately rather than after an auto update refresh.

![image](https://github.com/cmss13-devs/cmss13/assets/76988376/04a86193-b192-482f-a499-99857e9600dc)

However, I also tested this and the issue still occurs somereason where the spooling UI appears up to a second later than it should.

Reply:
because it's relying on the autoupdate still, launch_home blocks until the template is loaded

Reply:
Why not CallAsync then?

Reply:
> Why not CallAsync then?

then we wouldn't end up hitting the spooling screen - instead, launch_home() just invokes a tgui refresh when spooling is set to TRUE